server/services/applicationService.ts

```diff
@@ -56,18 +56,29 @@ export default class ApplicationService {
     return page
   }
 
-  save(page: TasklistPage, request: Request) {
+  async save(page: TasklistPage, request: Request) {
     const errors = page.errors ? page.errors() : []
+
     if (errors.length) {
       throw new ValidationError(errors)
     } else {
-      request.session = this.fetchOrInitializeSessionData(request.session, request.params.task, request.params.id)
-
-      request.session.application[request.params.id][request.params.task][request.params.page] = page.body
-      request.session.previousPage = page.name
+      this.saveToSession(page, request)
+      await this.saveToApi(request)
     }
   }
 
+  private saveToSession(page: TasklistPage, request: Request) {
+    request.session = this.fetchOrInitializeSessionData(request.session, request.params.task, request.params.id)
+    request.session.application[request.params.id][request.params.task][request.params.page] = page.body
+    request.session.previousPage = page.name
+  }
+
+  private async saveToApi(request: Request) {
+    const client = this.applicationClientFactory(request.user.token)
+
+    await client.update(request.session.application[request.params.id] as Application, request.params.id)
```

Comment:
Should this be updating only the `data` attribute of the application here?

Reply:
Yes - good spot